### ISSUE TITLE
Moving Brush when XAxis has ticks and interval={0} set leads to NaN errors

### ISSUE THREAD

**Issue by jackcannon** [recharts/recharts#3447]
- [ ] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/competent-forest-k11ts4?file=/src/App.tsx)

### Steps to reproduce
Move the brush traveller

### What is expected?
No javascript errors

### What is actually happening?
The following javascript errors:
* Warning: Received NaN for the `x1` attribute. If this is expected, cast the value to a string.
* Warning: Received NaN for the `x2` attribute. If this is expected, cast the value to a string.

| Environment | Info |
|---|---|
| Recharts | v2.4.3 |
| React | react 18.2.0 |
| System | Mac OS |
| Browser | Chrome |

---
This is only present when all three of these are present:
* `ticks` (on `XAxis`)
* `interval={0}` (on `XAxis`)
* `Brush`.

If any of the three are omitted, there are no errors, however, it would very useful to be able to keep all manual ticks visible when using a Brush.
There is initially no problem until you try to move the Brush travellers.

**Comment by ckifer** [comment]
Thanks for the repro, will look into it

**Comment by nikolasrieble** [comment]
Investigation so far: 

The CartesianAxis produces the finalTicks. 

```
    console.log('before', finalTicks);

    if (_.isFunction(ticksGenerator)) {
      finalTicks = ticks && ticks.length > 0 ? ticksGenerator(this.props) : ticksGenerator(noTicksProps);
    }

    console.log('after', finalTicks);
```

In this specific case, the output of final ticks is an object 

```
{
    coordinate: Nan,
    values: [...], 
    offset: 0,
}
```

The log message is caused by computing x values for rendering using the coordinate of Nan. 
This indicates that the error originates from the ticksGenerator.

**Comment by ckifer** [comment]
fixed by https://github.com/recharts/recharts/pull/3454

Will be released in next minor version

**Comment by ckifer** [comment]
Released in 2.5!

**Comment by jackcannon** [comment]
Works great! Thanks both!
